sdks/python/apache_beam/runners/portability/portable_stager.py

```diff
@@ -0,0 +1,86 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""A :class:`FileHandler` to work with :class:`ArtifactStagingServiceStub`.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from apache_beam.portability.api import beam_artifact_api_pb2
+from apache_beam.portability.api import beam_artifact_api_pb2_grpc
+from apache_beam.runners.portability.stager import Stager
+
+
+class PortableStager(Stager):
+  """An implementation of :class:`Stager` to stage files on
+  ArtifactStagingService.
+
+  The class keeps track of pushed files and commit manifest once all files are
+  uploaded.
```

Comment:
I am curious, do we plan to add retry logic here (perhaps in later PRs) if some part of staging falls through? Or does GPRC handle retries automatically?

Reply:
GRPC does not retry on connection errors. 
I am not planning to add retry as of now as user can simply resubmit the pipeline if it fails.
We can revisit it if it becomes an issue.